runtime/recipe/handle.js

```diff
@@ -183,12 +183,27 @@ class Handle {
     }
   }
 
+  // Options:
+  // - showUnresolved: if true, includes unresolved handle details if
+  //   handle is in fact unresolved and they're available.
+  // - stripManifestSessionFromImmediate: if true and the handle is
+  //   an immediate, strips the session id prefix from the handle id.
   toString(nameMap, options) {
     // TODO: type? maybe output in a comment
     let result = [];
     result.push(this.fate);
     if (this.id) {
-      result.push(`'${this.id}'`);
+      // TODO(wkorman): `Manifest._processRecipe` has a TODO currently to mark a
+      // handle as immediate. Until we've a cleaner way to identify, we just
+      // scrutinize the handle id for a clear marker. A sample immediate handle
+      // id looks like:
+      // '!130690574120708:manifest:./manifest.manifest::0:immediateB'
+      let outputId = this.id.toString();
+      if (options && options.stripManifestSessionFromImmediate &&
+          /:immediate[^:]+$/.test(this.id)) {
+        outputId = outputId.replace(/^![^:]+:/, '');
```

Comment:
I considered adding and using a static method in `Manifest` to strip this, since knowledge of the constructed handle id format resides there, but having `Handle` reference `Manifest`, even a static method, felt circular. Open to thoughts of course.